eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -116,7 +116,7 @@ def to_side(self):
 
     init: const? rate_complex 
     definition: def_param "=" number
-    rule: ((label)? side ARROW side ("@" rate)? (";" variable)?) | ((label)? side BI_ARROW side ("@" rate "|" rate )? (";" variable)?)
+    rule: ((label)? side ARROW side ("@" rate)? (";" variable)?) | ((label)? side BI_ARROW side ("@" rate "|" rate )? (";" variable)?) | ((label)? side REPLICATION_ARROW side ("@" rate)? (";" variable)?)
```

Comment:
```suggestion
    rule: ((label)? side arrow side ("@" rate)? (";" variable)?)
    arrow: SINGLE_ARROW | BI_ARROW | REPLICATION_ARROW
```

Consider simplifying it to this. Note you will need to define `SINGLE_ARROW` , and also, the parsed tree will look a little bit different (one extra level in the tree), so you need to reflect it on appropriate spots.

Reply:
I simplified the replication and single arrow in commit [23e1f27](https://github.com/sybila/eBCSgen/pull/107/commits/23e1f273e60d152f659191239524c65a43d61ad2), the bidirectional arrow is separate to solve the issue with rates.